packages/generators/utils/language.ts

```diff
@@ -36,23 +40,60 @@ export function getBabelPlugin(): IModuleRule {
 				],
 			],
 		},
-		test: `${new RegExp(/\.js$/)}`,
+		// tslint:enable: object-literal-sort-keys
+	};
+}
+
+export function getTypescriptLoader(): IModuleRule {
+	return {
+		// TODO migrate tslint
+		// tslint:disable: object-literal-sort-keys
+		test: "/\.tsx?$/",
+		loader: "'ts-loader'",
+		include: ["path.resolve(__dirname, 'src')"],
+		exclude: ["/node_modules/"],
+		// tslint:enable: object-literal-sort-keys
 	};
 }
 
 export default function language(self, langType) {
 	switch (langType) {
 		case LangType.ES6:
-			self.configuration.config.webpackOptions.module.rules.push(
-				getBabelPlugin(),
-			);
 			self.dependencies.push(
 				"babel-loader",
 				"@babel/core",
 				"@babel/preset-env",
 			);
+			self.configuration.config.webpackOptions.module.rules.push(
+				getBabelLoader(),
+			);
 			break;
+
 		case LangType.Typescript:
+			self.dependencies.push(
+				"typescript",
+				"ts-loader",
+			);
+			self.configuration.config.webpackOptions.module.rules.push(
+				getTypescriptLoader(),
+			);
+			self.configuration.config.webpackOptions.resolve = {
+				extensions: [ "'.tsx'", "'.ts'", "'.js'" ],
+			};
+
+			// Update the entry files extensions to .ts
+			const jsEntryOption = self.configuration.config.webpackOptions.entry;
+			const jsExtension = new RegExp("\.js(?!.*\.js)");
+			let tsEntryOption = {};
+			if (typeof jsEntryOption === "string") {
+				tsEntryOption = jsEntryOption.replace(jsExtension, ".ts");
+			} else if (typeof jsEntryOption === "object") {
+				Object.keys(jsEntryOption).map((entry) => {
```

Comment:
In this case `forEach` is a better fit. Using a `map` without assigning it to a variable could lead to memory leaks.